src/Buerokratt.Common.UnitTests/Dmr/DmrServiceTests.cs

```diff
@@ -14,76 +17,158 @@ namespace Buerokratt.Common.UnitTests.Dmr
 {
     public sealed class DmrServiceTests : IDisposable
     {
-        private static readonly DmrServiceSettings DefaultServiceConfig = new()
+        private readonly MockHttpMessageHandler _httpMessageHandler = new();
+        private readonly Mock<ILogger<DmrService>> _logger;
+
+        public DmrServiceTests()
         {
-            DmrApiUri = new Uri("https://dmr.fakeurl.com")
-        };
+            _logger = new Mock<ILogger<DmrService>>();
+            _ = _logger
+                .Setup(m => m.IsEnabled(It.IsAny<LogLevel>()))
+                .Returns(true);
+        }
 
-        private readonly MockHttpMessageHandler httpMessageHandler = new();
-        private readonly Mock<ILogger<DmrService>> logger = new();
+        private static ICentOpsService ConfigureCentOps()
+        {
+            var centOpsMock = new Mock<ICentOpsService>();
+            _ = centOpsMock.Setup(x => x.FetchParticipantsByType(ParticipantType.Dmr))
+                       .ReturnsAsync(new[]
+                       {
+                           new Participant
+                           {
+                               Host = "https://dmr.fakeurl.com",
+                               Name = "Dmr",
+                               Type = ParticipantType.Dmr,
+                               Id = "1"
+                           }
+                       });
+
+            return centOpsMock.Object;
+        }
 
         [Fact]
         public async Task ShouldCallDmrApiWithGivenRequestWhenRequestIsRecorded()
         {
-            _ = httpMessageHandler.SetupWithExpectedMessage();
+            // Arrange
+            _ = _httpMessageHandler.SetupWithExpectedMessage();
 
-            var clientFactory = GetHttpClientFactory(httpMessageHandler, DefaultServiceConfig);
+            var centOps = ConfigureCentOps();
+            var clientFactory = GetHttpClientFactory(_httpMessageHandler);
 
-            var sut = new DmrService(clientFactory.Object, DefaultServiceConfig, logger.Object);
+            var sut = new DmrService(centOps, clientFactory.Object, new DmrServiceSettings(), _logger.Object);
 
             sut.Enqueue(GetDmrRequest());
 
+            // Act
             await sut.ProcessRequestsAsync().ConfigureAwait(false);
 
-            httpMessageHandler.VerifyNoOutstandingExpectation();
+            // Assert
+            _httpMessageHandler.VerifyNoOutstandingExpectation();
+
+            _logger.Verify(
+               m => m.Log(
+                   LogLevel.Information,
+                   It.Is<EventId>(e => e.Id == 1 && e.Name == "DmrCallbackPosted"),
```

Comment:
Is there a `const` somewhere that could be used rather than a hard coded string?

Reply:
I've moved things around a bit here - should solve this problem